### PR TITLE
Remove one string allocation in XmlTextEncoder by using stackalloc span and TryFormat

### PR DESCRIPTION
**Pull request by TrayanZapryanov**
Replace several int.ToString("X", NumberFormatInfo.InvariantInfo) by using stackalloc char[] and Tryformat.

### CODE REVIEW

File: src/libraries/System.Private.Xml/src/System/Xml/Core/XmlTextEncoder.cs [hunk 1]
```diff
@@ -477,16 +475,17 @@ internal void WriteCharEntity(char ch)
                 throw new ArgumentException(SR.Xml_InvalidSurrogateMissingLowChar);
             }
 
-            string strVal = ((int)ch).ToString("X", NumberFormatInfo.InvariantInfo);
+            Span<char> chAsSpan = stackalloc char[8];
```
Comment (TrayanZapryanov) [inline src/libraries/System.Private.Xml/src/System/Xml/Core/XmlTextEncoder.cs:478]: I am not sure if 8 is the correct value here.

**Comment by stephentoub** [comment]
We had a _long_ discussion about this one in https://github.com/dotnet/runtime/pull/61774.

**Comment by TrayanZapryanov** [comment]
> We had a _long_ discussion about this one in #61774.

Okay I will close this PR, but let's discuss one idea that popup in my mind for this if you see some value in it.
What about creating a class which inherits TextWriter(also receives another TextWriter in ctor) and keeps inside small char[](1/2/3k). Then this class is used everywhere instead of current one. Biggest benefit of it will be that all inherited methods will write to this buffer and when it reach its maximum - flush it to the underlying writer either through it's write "string" or "char[]" or even ReadOnlySpan<char> methods.
Also this class can provide methods similar to ValueStringBuilder - at least giving us a span , so we can use it to format **value** types. From my previous checks of the code I see that all int/datetimes... are formatted as strings through XmlConvert which converts them to strings. Instead now it will write them to the Span provided from new class.

@stephentoub, @krwq

**Comment by krwq** [comment]
@TrayanZapryanov I'm ok with making any experimentation if you can see large perf potential benefit. Make sure to write some reasonable benchmarks covering all common scenarios. Technically such TextWriter doesn't really need any changes to the current code as you can technically use it in your own code so perhaps even if this finds useful for very narrow scenarios it might be more of a blog post rather than change and people mind find it useful. I'd suggest start with opening an issue explaining more the idea and if you got some time for experimentation go ahead and share your code and benchmarks used. Worst case scenario we will not find any benefit and not change anything and close the issue.

**Pull request closed without merging**